src/MoBi.Presentation/Presenter/EditIndividualParameterPresenter.cs

```diff
@@ -0,0 +1,154 @@
+      {
+         if (_individualParameter.Formula != null)
+            return;
+
+         var (command, explicitFormula) = _formulaTask.CreateNewFormulaInBuildingBlock(typeof(ExplicitFormula), _individualParameterDTO.Dimension, _buildingBlock.FormulaCache.Select(x => x.Name), _buildingBlock);
+
+         var macroCommand = new MoBiMacroCommand
+         {
+            ObjectType = new ObjectTypeResolver().TypeFor<IndividualParameter>(),
+            Description = AppConstants.Commands.SetValueAndFormula,
+            CommandType = AppConstants.Commands.AddCommand
+         };
+
+         macroCommand.Add(command);
+         macroCommand.Add(_interactionTasksForIndividualBuildingBlock.SetFormula(_buildingBlock, _individualParameter, explicitFormula));
+         AddCommand(macroCommand);
+
+         Edit(_individualParameter, _buildingBlock);
+      }
+
+      public void Edit(IndividualParameter individualParameter, IndividualBuildingBlock buildingBlock)
+      {
+         _buildingBlock = buildingBlock;
+         _individualParameter = individualParameter;
+         _individualParameterDTO = _individualParameterToIndividualParameterDTOMapper.MapFrom(individualParameter);
+         createDistributionValue();
+         _view.BindTo(_individualParameterDTO);
+         _editValueOriginPresenter.Edit(individualParameter);
+         _view.ShowWarningFor(buildingBlock?.Name);
+         updateFormulaView();
+      }
+      
+      private void createDistributionValue()
+      {
+         if (_individualParameter.DistributionType.HasValue)
+            _individualParameterDTO.DistributionValue = _pathAndValueEntityToDistributedParameterMapper.MapFrom(_individualParameter, _individualParameter.DistributionType.Value, subParametersFrom(_buildingBlock, _individualParameter)).Value;
+      }
+
+      private IReadOnlyList<IndividualParameter> subParametersFrom(IndividualBuildingBlock buildingBlock, IndividualParameter individualParameter)
+      {
+         return buildingBlock.Where(x => x.ContainerPath.Equals(individualParameter.Path)).ToList();
+      }
+
+      public void UpdateValue(double? newValue)
+      {
+         if (!newValue.HasValue || isStringEquivalent(_individualParameterDTO.Value, newValue)) 
+            return;
+         
+         AddCommand(_interactionTasksForIndividualBuildingBlock.SetValue(_buildingBlock, newValue, _individualParameter));
+         Edit(_individualParameter, _buildingBlock);
+      }
+
+      private bool isStringEquivalent(double? value, double? newValue)
+      {
+         if(value.HasValue && newValue.HasValue)
+            return value.ToString() == newValue.ToString();
+
+         return false;
+      }
+
+      public void UpdateUnit(Unit unit)
+      {
+         AddCommand(_interactionTasksForIndividualBuildingBlock.SetUnit(_buildingBlock, _individualParameter, unit));
+      }
+
+      public void NavigateToParameter()
+      {
+         _editTask.Edit(_buildingBlock);
```

Comment:
is building block the individual?  can we rename to individualBuildingBlock? (or individual)